CMS-Enterprise/sbom-harbor
# Update API generator and finish Team routes.

## Summary

This PR completes the endpoints for the Team service and updates the cargo-generate module to generate code for this design.

### Added

- New Team controller and service functions.
- Handler configuration to the `api::main` function for all `Team` routes.
- Basic validation for all `Team` controller functions.
- The `api/tests/test-api.http` file for running manual tests locally against debug builds.
- 

### Changed

- Moved the `openapi` directory to live under the `api` crate folder.
- Refactored the cargo-generate module to render `controllers` and `services`.

### Removed

Deleted some out of date test code.

## How to test

- Pull branch
- Make sure a local mongo container is running
- Run `cargo test`
- Optionally, you should be able to run the `api::main` function from the IDE and then use the new `http` test file to run manual tests or debug against the running instance.
